### DISCUSSION TITLE
Help! why the command line reminds that "Error, yalmiptest, ops=sdpsetting('solver',prefered_solver)"

### DISCUSSION THREAD

**Discussion by Shuyinghengxieshuiqingqian** [yalmip/YALMIP#1362] [General discussion]
when I input 'yalmiptest' to check if its installation is sucessful or not, the command line reminds me that "Error, yalmiptest, ops=sdpsetting('solver',prefered_solver)" . Strangely, 'sdpvar' works. Thanks for your advice in advance!

**Comment by johanlofberg** [comment]
Are you saying sdpsettings doesn't work, i.e. what happens when you run sdpsettings?

**Comment by Shuyinghengxieshuiqingqian** [comment]
Thanks for your reply! I took a screenshot as the following picture. As you see, I have already add 'Yalmip' to that toolbox for MATLABR2016a

However, when I input 'yalmiptest' into the command line of 'MATLABR2016a', it goes wrong and displays the following message.
I was wondering whether I installed 'YALMIP' successfully or not; If not, why 'sdpvar(1,2)' works? If it was installed, why it goes wrong when I input 'yalmiptest'?
Thank you very much! I am looking forward your reply. (*^_^*)






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yalmip/YALMIP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月15日(星期五) 下午3:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yalmip/YALMIP] Help! why the command line reminds that "Error, yalmiptest, ops=sdpsetting('solver',prefered_solver)" (Discussion #1362)





 
Are you saying sdpsettings doesn't work, i.e. what happens when you run sdpsettings?

**Reply by johanlofberg** [reply]
You did not supply a picture, nor did you say what the "following message" was

**Comment by Shuyinghengxieshuiqingqian** [comment]
Sorry, I sent those pictures in the attachment. Can you see that?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "疏浅"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月15日(星期五) 下午3:19
***@***.***&gt;;

主题:&nbsp;回复： [yalmip/YALMIP] Help! why the command line reminds that "Error, yalmiptest, ops=sdpsetting('solver',prefered_solver)" (Discussion #1362)



Thanks for your reply! I took a screenshot as the following picture. As you see, I have already add 'Yalmip' to that toolbox for MATLABR2016a

However, when I input 'yalmiptest' into the command line of 'MATLABR2016a', it goes wrong and displays the following message.
I was wondering whether I installed 'YALMIP' successfully or not; If not, why 'sdpvar(1,2)' works? If it was installed, why it goes wrong when I input 'yalmiptest'?
Thank you very much! I am looking forward your reply. (*^_^*)






------------------ 原始邮件 ------------------
发件人:                                                                                                                        "yalmip/YALMIP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月15日(星期五) 下午3:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yalmip/YALMIP] Help! why the command line reminds that "Error, yalmiptest, ops=sdpsetting('solver',prefered_solver)" (Discussion #1362)





 
Are you saying sdpsettings doesn't work, i.e. what happens when you run sdpsettings?

**Reply by johanlofberg** [reply]
No. It looks like you are using your email client. I think you will have to use the web browser directly in the thread

**Comment by Shuyinghengxieshuiqingqian** [comment]
I sent you those picutres using my another email address. Can you see it? If not, I copied those error messages in the command window of 'MATLAB R2016a' as the following:


&gt;&gt; yalmiptest
Error: File: sdpsettings.m Line: 1341 Column: 31
The input character is not valid in MATLAB statements or expressions.


Error in yalmiptest (line 132)
&nbsp; &nbsp; ops = sdpsettings('Solver',prefered_solver);
&nbsp;
&gt;&gt; sdpvar(1,2)
Linear matrix variable 1x2 (full, real, 2 variables)
Coefficient range: 1 to 1
&gt;&gt; sdpsettings
Error: File: sdpsettings.m Line: 1341 Column: 31
The input character is not valid in MATLAB statements or expressions.







------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yalmip/YALMIP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月15日(星期五) 下午3:27
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yalmip/YALMIP] Help! why the command line reminds that "Error, yalmiptest, ops=sdpsetting('solver',prefered_solver)" (Discussion #1362)





 
No. It looks like you are using your email client. I think you will have to use the web browser directly in the thread

**Reply by johanlofberg** [reply]
OK, now I see

There is an issue in the master branch where "" is used in sdpsettings which does not work in old versions of MATLAB. You will either have to replace all "" with '', or install the develop branch where this is fixed

**Comment by Shuyinghengxieshuiqingqian** [comment]
I can't thank you enough for your valuable advice!&nbsp;This error have bugged me for the whole day.&nbsp;
I hope you can visit China someday and I am definitely willing to serve as your tourist guide.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yalmip/YALMIP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年12月15日(星期五) 下午3:59
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yalmip/YALMIP] Help! why the command line reminds that "Error, yalmiptest, ops=sdpsetting('solver',prefered_solver)" (Discussion #1362)





 
OK, now I see
 
There is an issue in the master branch where "" is used in sdpsettings which does not work in old versions of MATLAB. You will either have to replace all "" with '', or install the develop branch where this is fixed
